src/Workspaces/Core/Portable/NamingStyles/EditorConfig/EditorConfigNamingStyleParser.cs

```diff
@@ -59,10 +60,28 @@ public static NamingStylePreferences ParseDictionary(IReadOnlyDictionary<string,
                 }
             }
 
-            return new NamingStylePreferences(
+            var preferences = new NamingStylePreferences(
                 symbolSpecifications.ToImmutableAndFree(),
                 namingStyles.ToImmutableAndFree(),
                 namingRules.ToImmutableAndFree());
+
+            var orderedRules = preferences.Rules.NamingRules
+                .OrderBy(rule => rule, NamingRuleAccessibilityListComparer.Instance)
+                .ThenBy(rule => rule, NamingRuleModifierListComparer.Instance)
+                .ThenBy(rule => rule, NamingRuleSymbolListComparer.Instance)
+                .ThenBy(rule => rule.NamingStyle.Name, StringComparer.OrdinalIgnoreCase)
+                .ThenBy(rule => rule.NamingStyle.Name, StringComparer.Ordinal);
```

Comment:
📝 The final two rules to order by name increase the chances that we will change behavior relative to the old strict ordering by position in the .editorconfig file(s), but omitting them introduces non-determinism in the build.

Reply:
Comment should be in code.